apps/condo/domains/contact/components/ContactsEditor.tsx

```diff
@@ -0,0 +1,556 @@
+import { useOrganization } from '@core/next/organization'
+import { ErrorsWrapper } from '@condo/domains/ticket/components/BaseTicketForm/ErrorsContainer'
+
+interface ILabelsProps {
+    left: React.ReactNode,
+    right?: React.ReactNode,
+}
+
+/**
+ * Displays validation error, but hides form input
+ */
+const ErrorContainerOfHiddenControl = styled.div`
+  .ant-form-item-control-input {
+    display: none;
+  }
+`
+
+const LABELS_COL_STYLE = { marginTop: '24px' }
+
+const Labels: React.FC<ILabelsProps> = ({ left, right }) => (
+    <>
+        <Col span={10} style={LABELS_COL_STYLE}>
+            <Typography.Text type="secondary">
+                {left}
+            </Typography.Text>
+        </Col>
+        <Col span={10} style={LABELS_COL_STYLE}>
+            <Typography.Text type="secondary">
+                {right}
+            </Typography.Text>
+        </Col>
+        <Col span={2} style={LABELS_COL_STYLE}>
+        </Col>
+        <Col span={2} style={LABELS_COL_STYLE}>
+        </Col>
+    </>
+)
+
+type ContactFields = {
+    name: string,
+    phone: string,
+}
+
+type ContactValue = ContactFields & {
+    id?: string,
+}
+
+interface IContactEditorProps {
+    form: FormInstance<any>,
+    // Customizeable field names of the provided `form`, where editor component will be mounted
+    // Fields `clientName` and `clientPhone` are not hardcoded to make this component
+    // usable in any form, where contact information fields may be different.
+    // Also, this makes usage of the component explicitly, — it's clear, what fields will be set.
+    fields: {
+        id: string,
+        phone: string,
+        name: string,
+    },
+    value?: ContactValue,
+    onChange: (contact: ContactFields, isNew: boolean) => void,
+    // Contacts for autocomplete will be fetched for specified organization
+    organization?: string,
+    // Contacts for autocomplete will be fetched for specified property
+    property?: string,
+    // Contacts for autocomplete will be fetched for specified unit of the property
+    unitName?: string,
+}
+
+interface IContactsEditorHookArgs {
+    // Organization scope for contacts autocomplete and new contact, that can be created
+    organization: string,
+}
+
+interface IContactsEditorHookResult {
+    createContact: (organization: string, property: string, unitName: string) => Promise<void>,
+    ContactsEditorComponent: React.FC<IContactEditorProps>,
+}
+
+export const useContactsEditorHook = ({ organization }: IContactsEditorHookArgs): IContactsEditorHookResult => {
+    // Field value will be initialized only on user interaction.
+    // In case of no interaction, no create action will be performed
+    const [contactFields, setContactFields] = useState({})
+    const [shouldCreateContact, setShouldCreateContact] = useState(false)
+
+    // Closure of `createContact` will be broken, when it will be assigned to another constant outside of this hook
```

Comment:
redurant comment

Reply:
I don't think so, because I've described, why refs are used here. Not everyone will quickly figure out, what side-effects this implementation solves.